src/nvim/ex_cmds.c

```diff
@@ -2546,11 +2546,17 @@ int do_ecmd(
     }
     check_arg_idx(curwin);
 
-    // If autocommands change the cursor position or topline, we should keep
-    // it.  Also when it moves within a line.
+    // If autocommands change the cursor position or topline, we should
+    // keep it.  Also when it moves within a line. But not when it moves
+    // to the first non-blank.
     if (!equalpos(curwin->w_cursor, orig_pos)) {
-      newlnum = curwin->w_cursor.lnum;
-      newcol = curwin->w_cursor.col;
+      const char_u *text = get_cursor_line_ptr();
```

Comment:
We should not have renamed nor moved `ml_get_curline` and `ml_get_cursor`. The new name and location in cursor.c makes little sense.